### PR TITLE
Make argument constant as optional in proof generation.

### PR DESCRIPTION
**Pull request by VolodymyrBabaiev**

### CODE REVIEW

File: src/zendoo/mcTestCall.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <iostream>
 #include <cassert>
 #include <string>
+#include "gtest/libzendoo_test_files.h"
```
Comment (DanieleDiBenedetto) [inline src/zendoo/mcTestCall.cpp:8]: Why do you need this ?

**Review by DanieleDiBenedetto — COMMENTED** [review]
Would be nice if you avoid specifying the constant in some of the python tests just to see that everything is working properly

File: src/zendoo/mcTestCall.cpp (hunk 1 above)
Comment (VolodymyrBabaiev) [reply src/zendoo/mcTestCall.cpp:8]: I took SAMPLE_FIELD from this file for the default value.

Comment (DanieleDiBenedetto) [reply src/zendoo/mcTestCall.cpp:8]: But you can safely pass NULL to the zendoo_create_cert_test_proof function

File: src/zendoo/mcTestCall.cpp [hunk 2]
```diff
@@ -87,12 +88,19 @@ void create_verify_test_cert_proof(std::string ps_type_raw, int argc, char** arg
...
+
+    field_t* constant_f = NULL;
+    if (constant_present > 0) {
+        assert(IsHex(argv[arg]));
+        auto constant = ParseHex(argv[arg++]);
+        assert(constant.size() == 32);
+        constant_f = zendoo_deserialize_field(constant.data(), &ret_code);
+        assert(ret_code == CctpErrorCode::OK);
+    } else {
+        constant_f = zendoo_deserialize_field(SAMPLE_FIELD.data(), &ret_code);
```
Comment (DanieleDiBenedetto) [inline src/zendoo/mcTestCall.cpp:101]: You should pass NULL if constant is not present

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

File: src/zendoo/mcTestCall.cpp (hunk 2 above)
Comment (i-Alex) [reply src/zendoo/mcTestCall.cpp:101]: agree with Daniele

**Review by i-Alex — APPROVED** [review]

**Review by DanieleDiBenedetto — APPROVED** [review]

**Review by DanieleDiBenedetto — APPROVED** [review]

**Review by i-Alex — APPROVED** [review]

File: src/zendoo/mcTestCall.cpp [hunk 3]
```diff
@@ -9,11 +9,11 @@
...
+ *  1) ./mcTest "generate" "cert/cert_no_const/csw" "darlin/cob_marlin" "params_dir" "segment_size" "num_constraints"
  *  Generates SNARK pk and vk for a test cert/csw circuit using darlin/coboundary_marlin proving system;
  *
- *  2) ./mcTest "create" "cert" "darlin/cob_marlin" <"-v"> <"-zk"> "proof_path" "params_dir" "segment_size"
- *  "sc_id" "epoch_number" "quality" "constant" "end_cum_comm_tree_root", "btr_fee", "ft_min_amount" "num_constraints"
+ *  2) ./mcTest "create" "cert/cert_no_const" "darlin/cob_marlin" <"-v"> <"-zk"> "proof_path" "params_dir" "segment_size"
+ *  "sc_id" "epoch_number" "quality" "constant_presence" ["constant"] "end_cum_comm_tree_root", "btr_fee", "ft_min_amount" "num_constraints"
  *  "bt_list_len", "pk_dest_0" "amount_0" "pk_dest_1" "amount_1" ... "pk_dest_n" "amount_n",
  *  "custom_fields_list_len", "custom_field_0", ... , "custom_field_1"
  *  Generates a TestCertificateProof;
```
Comment (DanieleDiBenedetto) [inline src/zendoo/mcTestCall.cpp:19]:
```suggestion
 *  2) ./mcTest "create" "cert/cert_no_const" "darlin/cob_marlin" <"-v"> <"-zk"> "proof_path" "params_dir" "segment_size"
 *  "sc_id" "epoch_number" "quality" ["constant"] "end_cum_comm_tree_root", "btr_fee",
 *  "ft_min_amount" "num_constraints"
 *  "bt_list_len", "pk_dest_0" "amount_0" "pk_dest_1" "amount_1" ... "pk_dest_n" "amount_n",
 *  "custom_fields_list_len", "custom_field_0", ... , "custom_field_1"
 *  Generates a TestCertificateProof.
 *  NOTE: "constant" param must be present if "cert" has been passed; If "cert_no_const" has been passed,
 *        instead, "constant" param must not be present.
```

File: src/zendoo/mcTestCall.cpp [hunk 4]
```diff
@@ -448,6 +452,9 @@ void create_verify(int argc, char** argv)
     if (circ_type_raw == "cert") {
         assert(argc >= 17);
         create_verify_test_cert_proof(ps_type_raw, argc, argv);
+    } else if (circ_type_raw == "cert_no_const") {
+        assert(argc >= 16);
+        create_verify_test_cert_proof(ps_type_raw, argc, argv);
     } else if (circ_type_raw == "csw") {
         create_verify_test_csw_proof(ps_type_raw, argc, argv);
```
Comment (DanieleDiBenedetto) [inline src/zendoo/mcTestCall.cpp:459]: Let's take the occasion also here to assert correct size of params for csw, just for completeness. It should be:

`assert(argc >= 13 && argc <= 16);`

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

File: src/zendoo/mcTestCall.cpp (hunk 4 above)
Comment (VolodymyrBabaiev) [reply src/zendoo/mcTestCall.cpp:463]: Done.

File: src/zendoo/mcTestCall.cpp [hunk 5]
```diff
@@ -18,6 +18,9 @@
  *  "custom_fields_list_len", "custom_field_0", ... , "custom_field_1"
  *  Generates a TestCertificateProof;
  *
+ *  NOTE: "constant" param must be present if "cert" has been passed; If "cert_no_const" has been passed,
```
Comment (DanieleDiBenedetto) [inline src/zendoo/mcTestCall.cpp:21]: There is still "constant_presence" param in the comment, it must be removed

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

File: src/zendoo/mcTestCall.cpp [hunk 6]
```diff
@@ -18,8 +18,6 @@
  *  "custom_fields_list_len", "custom_field_0", ... , "custom_field_1"
  *  Generates a TestCertificateProof;
  *
- *  NOTE: "constant" param must be present if "cert" has been passed; If "cert_no_const" has been passed,
```
Comment (DanieleDiBenedetto) [inline src/zendoo/mcTestCall.cpp]: No lol this comment was ok. In line 16 there's still "constant_presence" param. That one must be removed

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

**Review by DanieleDiBenedetto — APPROVED** [review]

**Review by cronicc — CHANGES_REQUESTED** [review]
Conflicts with #139

A branch on zendoo_mc_cryptolib with both https://github.com/HorizenOfficial/zendoo-mc-cryptolib/pull/45 and https://github.com/HorizenOfficial/zendoo-mc-cryptolib/pull/46 merged and conflicts resolved is needed.

**Pull request merged**